File: src/main/java/com/gluck/jobtracker/service/ApplicationMapper.java
package com.gluck.jobtracker.service;

import com.gluck.jobtracker.repository.JobApplicationEntity;
import com.gluck.jobtracker.model.JobApplicationRequest;
import com.gluck.jobtracker.model.JobApplicationResponse;
import org.springframework.stereotype.Component;

@Component
public class ApplicationMapper {

    public JobApplicationEntity toEntity(JobApplicationRequest request) {
        JobApplicationEntity entity = new JobApplicationEntity();
        entity.setPosition(request.getPosition());
        entity.setCompanyName(request.getCompanyName());
        entity.setStatus(request.getStatus());
        entity.setDateApplied(request.getDateApplied());
        entity.setDescription(request.getDescription());
        return entity;
    }

    public JobApplicationResponse toResponse(JobApplicationEntity application) {
        return new JobApplicationResponse(
            application.getId(),
            application.getPosition(),
            application.getCompanyName(),
            application.getStatus(),
            application.getDateApplied(),
            application.getDescription()
        );
    }

    public JobApplicationRequest toRequest(JobApplicationEntity application) {
        return new JobApplicationRequest(
            application.getPosition(),
            application.getCompanyName(),
            application.getStatus(),
            application.getDateApplied(),
            application.getDescription()
        );
    }
}

File: src/main/kotlin/com/gluck/jobtracker/service/ApplicationMapper.kt
package com.gluck.jobtracker.service

import com.gluck.jobtracker.model.JobApplicationEntity
import com.gluck.jobtracker.model.JobApplicationRequest
import com.gluck.jobtracker.model.JobApplicationResponse
import org.springframework.stereotype.Component

@Component
class ApplicationMapper {

    fun toEntity(request: JobApplicationRequest): JobApplicationEntity {
        return JobApplicationEntity(
            position = request.position,
            companyName = request.companyName,
            status = request.status,
            dateApplied = request.dateApplied,
            description = request.description
        )
    }

    fun toResponse(application: JobApplicationEntity): JobApplicationResponse {
        return JobApplicationResponse(
            id = application.id,
            position = application.position,
            companyName = application.companyName,
            status = application.status,
            dateApplied = application.dateApplied,
            description = application.description
        )
    }

    fun toRequest(application: JobApplicationEntity): JobApplicationRequest {
        return JobApplicationRequest(
            position = application.position,
            companyName = application.companyName,
            status = application.status,
            dateApplied = application.dateApplied,
            description = application.description
        )
    }

}